eZ/Publish/Core/IO/IOMetadataHandler/LegacyDFSCluster.php

```diff
@@ -291,4 +283,33 @@ protected function mapSPIBinaryFileCreateStructToSPIBinaryFile(SPIBinaryFileCrea
 
         return $spiBinaryFile;
     }
+
+        /**
+     * @return string
+     * generate the correct SQL query to insert data for mysql and postgres
+     */
+    protected function getInsertQuery() {
+        $query = <<<SQL
+INSERT INTO ezdfsfile
+(name, name_hash, name_trunk, mtime, size, scope, datatype)
+VALUES (:name, :name_hash, :name_trunk, :mtime, :size, :scope, :datatype)
+SQL;
+
+        $mysqlDuplicateConstraint = <<<SQL
+ON DUPLICATE KEY UPDATE 
+datatype=VALUES(datatype), scope=VALUES(scope), size=VALUES(size),
+mtime=VALUES(mtime)
+SQL;
+
+        $pgDulplicateConstraint = <<<SQL
```

Comment:
Given this is the else part, and there might be more DBs to support in the future, is it ansi compatible? If so maybe rename variable name

Reply:
...and fix the speling error while we're at it.

Reply:
The ANSI way is apparently MERGE INTO. Postgres seems to support that. So then we should probably rename as you said, and rewrite to use merge syntax.
https://stackoverflow.com/questions/2662425/ansi-sql-question-how-to-insert-or-update-a-record-if-it-already-exists
https://www.postgresql.org/message-id/attachment/23520/sql-merge.html

Reply:
Odd that the DBAL/Connection driver can be `null` though, as in the text. Do we know what can cause that?